examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py

```diff
@@ -100,13 +115,25 @@ def _get_item(self, idx: int) -> Tuple[Any, ...]:
                 data.update({"caption": td["text_emb"], "mask": td["mask"]})
 
         if self._vae_latent_folder:
-            if "fps" not in data:  # cache FPS for further iterations
+            # pick a resolution randomly if there are multi-resolution latents in vae folder
+            vae_latent_path = data["vae_latent"]
+            if self.num_latent_resolution > 1:
+                ridx = random.randint(0, self.num_latent_resolution - 1)
+                vae_latent_path = vae_latent_path.replace(self._vae_latent_folder, self.latent_resolution_prefix[ridx])
+            # print("D--: vae latent npz: ", vae_latent_path)
+
+            vae_latent_data = np.load(vae_latent_path)
+
+            # get fps from csv, or cached latents, or from original video in order
+            if "fps" in data:  # cache FPS for further iterations
+                data["fps"] = np.array(data["fps"], dtype=np.float32)
+            elif "fps" in vae_latent_data:
+                data["fps"] = np.array(vae_latent_data["fps"], dtype=np.float32)
+            else:
                 with VideoReader(data["video"]) as reader:
                     data["fps"] = self._data[idx]["fps"] = reader.fps
```

Comment:
`data["fps"]` should be a `float32` array

Reply:
fixed